Extensions/P2P/B_p2ptools.ts

```diff
@@ -7,43 +7,101 @@ namespace gdjs {
      */
     export namespace p2p {
       /**
-       * The peer to peer configuration.
+       * The type of the data that is sent across peerjs.
        */
-      let peerConfig: Peer.PeerJSOption = { debug: 1 };
+      type NetworkEvent = {
+        eventName: string;
+        data: string;
+      };
+
+      const isValidNetworkEvent = (event): event is NetworkEvent =>
+        typeof event.eventName === 'string' || typeof event.data === 'string';
 
       /**
-       * The p2p client.
+       * The data bound to an event that got triggered.
        */
-      let peer: Peer | null = null;
+      class EventData {
+        constructor(data: string, sender: string) {
+          this.data = data;
+          this.sender = sender;
+        }
+
+        /**
+         * The data sent alongside the event.
+         */
+        public readonly data: string = '';
+
+        /**
+         * The ID of the sender of the event.
+         */
+        public readonly sender: string = '';
+      }
 
       /**
-       * All connected p2p clients, keyed by their ID.
+       * An event that can be listened to.
        */
-      const connections: Record<string, Peer.DataConnection> = {};
+      class Event {
+        private data: EventData[] = [];
+        public dataloss = false;
+
+        /**
+         * Returns true if the event is triggered.
+         */
+        isTrigerred() {
+          return this.data.length > 0;
+        }
+
+        /**
+         * Add new data, to be called with the event data each time the event is triggered.
+         */
+        pushData(newData: EventData) {
+          if (this.dataloss && this.data.length > 0) return;
+          else this.data.push(newData);
+        }
+
+        /**
+         * Deleted the last event data, to be called when it is sure the event was processed throughly.
+         */
+        popData() {
+          this.data.shift();
+        }
+
+        /**
+         * Get the data sent with the last event triggering.
+         */
+        getEventData() {
```

Comment:
nitpicking: as we're already inside `Event` class, you should be able to call this just `getData`.
It will be "obvious" to the caller that this is the "event data", because the calling code will be something like `getEvent(...).getData()`.

Also it's consistent with push/popData.

Also do this for getEventSender => getSender :)